whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/processor/UpdateNotificationFileProcessor.java

```diff
@@ -74,67 +76,69 @@ public void processFile(){
-                LOGGER.info("The local version cannot be higher than the update notification version {}", source);
-                snapshotImporter.truncateTables();
-                return;
-            }
+                if (nrtmClientLastVersionInfo != null && !nrtmClientLastVersionInfo.sessionID().equals(updateNotificationFile.getSessionID())) {
+                    LOGGER.warn("Different session");
+                    throw new IllegalStateException("Different session");
+                }
 
-            if (nrtmClientLastVersionInfo != null && nrtmClientLastVersionInfo.version().equals(updateNotificationFile.getVersion())) {
-                LOGGER.info("There is no new version associated with the source {}", source);
-                return;
-            }
+                if (nrtmClientLastVersionInfo != null && nrtmClientLastVersionInfo.version() > updateNotificationFile.getVersion()) {
+                    LOGGER.warn("The local version cannot be higher than the update notification version {}", source);
+                    throw new IllegalStateException(String.format("The local version cannot be higher than the update notification version %s", source));
+                }
 
-            try {
-                if (nrtmClientLastVersionInfo == null) {
-                    snapshotImporter.doImport(source, updateNotificationFile.getSessionID(), updateNotificationFile.getSnapshot());
+                if (nrtmClientLastVersionInfo != null && nrtmClientLastVersionInfo.version().equals(updateNotificationFile.getVersion())) {
+                    LOGGER.info("There is no new version associated with the source {}", source);
+                    return;
                 }
 
-                final List<UpdateNotificationFileResponse.NrtmFileLink> newDeltas = getNewDeltasFromNotificationFile(source, updateNotificationFile);
-                deltaImporter.doImport(source, updateNotificationFile.getSessionID(), newDeltas);
-                persistUpdateFileVersion(source, updateNotificationFile, hostname);
-            } catch (Exception ex){
-                LOGGER.error("Failed to mirror database, cleaning up the tables", ex);
-                cleanUpTablesSafely();
-            }
-        });
+                try {
+                    if (nrtmClientLastVersionInfo == null) {
+                        snapshotImporter.doImport(source, updateNotificationFile.getSessionID(), updateNotificationFile.getSnapshot());
+                    }
+
+                    final List<UpdateNotificationFileResponse.NrtmFileLink> newDeltas = getNewDeltasFromNotificationFile(source, updateNotificationFile);
+                    deltaImporter.doImport(source, updateNotificationFile.getSessionID(), newDeltas);
+                    persistUpdateFileVersion(source, updateNotificationFile, hostname);
+                } catch (Exception ex) {
+                    LOGGER.error("Failed to mirror {} database, cleaning up the tables", source, ex);
```

Comment:
Move ERROR statement down to catch block below

Reply:
I put this error there because I wanted to know the source that cause the error.
The below bock is out of the loop so I cannot know the source there

Reply:
OK